docs: new tutorial and updates related to RoundRequestor. (#54)
algorand/conduit: docs/tutorials/IndexerWriter.md

# Conduit Indexer Writer

This document explains how to configure and deploy Conduit to populate a

> PostgreSLQ DB suitable to serve as a backend for the Indexer API.

Reviewer: Can we also link to the Indexer API?

Note: Windows is not currently supported by algod.

> 

Author: Adding a per-requisites section

The components you will configure and deploy are:
* Algod configured as a blockchain follower.
* PostgreSQL database.

> * Conduit configured to pull data from algod and push it to PostgreSQL

Reviewer: ```suggestion
* a Conduit pipeline configured to pull data from algod and push it to the PostgreSQL database.
```

```mermaid
graph LR;
    algod["algod"]
    pe["postgresql Exporter"]
    algodimp["algod Importer"]
    db["PostgreSQL DB"]
    indexer["Indexer REST API Service"]
    algod-->algodimp
    subgraph "Conduit Pipeline"
        algodimp-->pe;
    end
    pe-->db;
    db-->indexer
```

## Algod

> To source data from the blockchain algod is used. It will be controlled by

Reviewer: ```suggestion
The algod is used to source data from the blockchain. It will be controlled by
```

> Conduit in order to run synchronously. In this way it is able to store a

Reviewer: I don't have a good alternative suggestion, but the word "synchronously" on its own doesn't mean anything to me here.

> Conduit in order to run synchronously. In this way it is able to store a

Reviewer: ```suggestion
Conduit in order to run synchronously. In this way, the algod is able to store a
```

Author: Like the batman? Is this how we usually talk about algod?

Reviewer: lol. Hm sometimes we do, like to refer to a specific instance? and the "d" is for daemon, right? I would "the" a daemon. But also I don't care about that for this particular comment, just wanted to be specific instead of "it" here so it's not confused with Conduit.

minimal subset of data required for streaming data through Conduit.

We will cover a basic installation using `update.sh`, for [additional
information refer to the developer portal.](node-install-doc)

> ```bash

Reviewer: perhaps put spaces between the commands, so it's more obvious which command each set of comments belong to?

# download update.sh
wget https://raw.githubusercontent.com/algorand/go-algorand/rel/stable/cmd/updater/update.sh
# Select mainnet genesis file.
# Options: mainnet, testnet, betanet, devnet
NETWORK=mainnet

> # download binaries and initialize data directory.

Reviewer: ```suggestion
# download algod binaries and initialize data directory.
```

update.sh -n -i -c stable -p bin -d algod_data -g $NETWORK
# configure node to work with conduit
./bin/algocfg profile set conduit -d algod_data
```

At this stage you have all of the algod tools in `bin` and a `algod_data`
directory configured for Conduit with the desired network.

Start the node with `./bin/goal node start -d algod_data`

## PostgreSQL

This document cannot go into the complete details for configuring your
database. PostgreSQL or compatible database like Amazon Aurora are suitable
for the Indexer API. The [Indexer](indexer-readme) documentation more details.

For this tutorial we'll simplify things with a local docker image. This will
launch a container:
```bash

> docker run -d --name local-database -p 5555:5432 -e POSTGRES_PASSWORD=pgpass -e POSTGRES_USER=algorand -e POSTGRES_DB=conduit postgres

Author: Yes. Adding a comment at the top that this is not a production deployment.

* Docker is used for the database, so the `docker` command is required.

## Algod

The blockchain data source. Algod is controlled by Conduit to ensure the two
services are always operating on the same block. In this way algod is able to
store a minimal subset of blockchain data required for streaming data through
Conduit.

We will cover a basic installation using `update.sh`, for [additional
information refer to the developer portal.](node-install-doc)

```bash
# download update.sh
wget https://raw.githubusercontent.com/algorand/go-algorand/rel/stable/cmd/updater/update.sh

# Define which network to run on. This selects the correct genesis file. 
# Options: mainnet, testnet, betanet, devnet
NETWORK=mainnet

# download algod binaries and initialize data directory.
update.sh -n -i -c stable -p bin -d algod_data -g $NETWORK

# configure node to work with conduit
./bin/algocfg profile set conduit -d algod_data
```

At this stage you have all of the algod tools in `bin` and an `algod_data`
directory configured for Conduit with the desired network.

Start the node:
```bash

> ~$ ./bin/goal node start -d algod_data

Reviewer: ```suggestion
./bin/goal node start -d algod_data
```
to stay consistent with the above?

Author: Thanks. This is the format I usually use, but for some reason above I wrote it as though you would run a script.
